src/ILCompiler.Compiler/src/Compiler/JitHelper.cs

```diff
@@ -169,29 +169,53 @@ public static void GetEntryPoint(TypeSystemContext context, ReadyToRunHelper id,
                     break;
 
                 case ReadyToRunHelper.Mod:
-                    mangledName = "RhpIMod";
+                    if (context.Target.Architecture == TargetArchitecture.ARMEL)
+                        methodDesc = context.GetHelperEntryPoint("MathHelpers", "IMod");
+                    else
+                        mangledName = "RhpIMod";
```

Comment:
Using the RhpIMod helper for `ReadyToRunHelper.Mod` is never correct. Also, it should be used for ARM as well.

I think you can just return `methodDesc = context.GetHelperEntryPoint("MathHelpers", "IMod")` unconditionally here. In the unlikely case some other platform starts calling into this helper, it will be easy to see that the helper needs implemented because of the compilation will start failing.